### ISSUE TITLE
[split-button] pre-spec research

### ISSUE THREAD

**Issue by yinonov** [openui/open-ui#408]
# split-Button - spec

> A split button is a dual-function menu button that offers a default action as well as the possibility of choosing a different action by selecting from a set of alternatives.

_[Nielsen Norman Group](https://www.nngroup.com/articles/split-buttons/)_

Moreover, split button can be specified as 2 different behaviors:

1. as defined by NNG 👆 set of alternatives set the main action
2. set of alternatives are eagerly invoked actions as [presented here](https://web.dev/building-a-split-button-component)

### Implementation details:
the button UI is composed of 2 buttons:

1. main action
2. popup trigger chevron like suffixed button

![opensource adobe com_spectrum-web-components_components_split-button](https://user-images.githubusercontent.com/15140652/135709984-5b964f8e-0f52-4d97-acbf-1258b9dc2361.png)

- should they both be included in the tabbing navigation order or align to roving pattern?

- Would focus alone on chevron suffixed button open the popup menu?

- Keep state of where the focus was if the menu closes and opens back?

Some of the points raised and details will really require 1 finite defined behavior [mentioned above](#split-Button)

argyleink/gui-challenges#63 already touches the topic

**Comment by yinonov** [comment]
Got topic nowadays. For starters, is it safe to say that the discussion clearly sees the popup options as setting the main action and not immediately invoke an action?

**Comment by rajsite** [comment]
Are there established patterns for the aria of split buttons today in web?
It does not look like there is an existing aria pattern: https://github.com/w3c/aria-practices/issues/2316
But curious if design systems have a common convention that works well with users.

**Comment by chrisdholt** [comment]
> Are there established patterns for the aria of split buttons today in web? It does not look like there is an existing aria pattern: [w3c/aria-practices#2316](https://github.com/w3c/aria-practices/issues/2316) But curious if design systems have a common convention that works well with users.

I don’t think this has any common pattern, but I’m a big proponent that these are TWO buttons and this two tab stops. Button has a standard element and pattern, split button is a layout that logically and presentationally groups these, often a menu button (pattern) and a button (pattern).. I’m pretty averse to the “one tab stop” approach because it implies two buttons can be one.

(There’s my strong opinion for the day 😁)

**Comment by bramus** [comment]
I recently built a split button which triggered the question if this was something that is being tackled at Open UI or not, which led me to this issue.

The way I built it:
- The button and the chevron are individual tabstops.
- Activating the chevron opens and focuses the popover, allowing you to then cycle through the options using the arrow keys.
- It is only after activating an option that the popover popover and the the button’s text updates. The focus remains on the chevron.
- Activating the button performs the button’s action.

Here’s a recording:

https://github.com/user-attachments/assets/e2a135e7-5dcf-4d25-911f-28254eba7fef
